modules/apps/content-dashboard/content-dashboard-document-library-test/src/testIntegration/java/com/liferay/content/dashboard/document/library/internal/item/action/provider/test/ViewFileVersionContentDashboardItemVersionActionProviderTest.java

```diff
@@ -0,0 +1,176 @@
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.mock.web.MockHttpServletRequest;
+
+/**
+ * @author Jürgen Kappler
+ */
+@RunWith(Arquillian.class)
+public class ViewFileVersionContentDashboardItemVersionActionProviderTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup(
+			TestPropsValues.getCompanyId(), TestPropsValues.getUserId(), 0);
+	}
+
+	@Test
+	public void testGetContentDashboardItemAction() throws Exception {
+		_fileEntry = _dlAppLocalService.addFileEntry(
+			RandomTestUtil.randomString(), TestPropsValues.getUserId(),
+			_group.getGroupId(), DLFolderConstants.DEFAULT_PARENT_FOLDER_ID,
+			RandomTestUtil.randomString() + "." + ContentTypes.IMAGE_JPEG,
+			MimeTypesUtil.getExtensionContentType(ContentTypes.IMAGE_JPEG),
+			new byte[0], null, null,
+			ServiceContextTestUtil.getServiceContext(_group.getGroupId()));
+
+		MockHttpServletRequest mockHttpServletRequest =
+			new MockHttpServletRequest();
+
+		MockLiferayResourceRequest mockLiferayResourceRequest =
+			new MockLiferayResourceRequest();
+
+		mockLiferayResourceRequest.setAttribute(
+			WebKeys.THEME_DISPLAY, _getThemeDisplay(mockHttpServletRequest));
+
+		mockHttpServletRequest.setAttribute(
+			JavaConstants.JAVAX_PORTLET_REQUEST, mockLiferayResourceRequest);
+
+		mockHttpServletRequest.setAttribute(
+			JavaConstants.JAVAX_PORTLET_RESPONSE,
+			new MockLiferayPortletRenderResponse());
+
+		ContentDashboardItemVersionAction contentDashboardItemVersionAction =
+			_contentDashboardItemVersionActionProvider.
+				getContentDashboardItemVersionAction(
+					_fileEntry.getFileVersion(), mockHttpServletRequest);
+
+		Assert.assertEquals(
+			"view", contentDashboardItemVersionAction.getName());
+		Assert.assertEquals(
+			"view", contentDashboardItemVersionAction.getIcon());
+		Assert.assertEquals(
+			_language.get(LocaleUtil.US, "view"),
+			contentDashboardItemVersionAction.getLabel(LocaleUtil.US));
+
+		String url = contentDashboardItemVersionAction.getURL();
+
+		Assert.assertTrue(Validator.isNotNull(url));
+
+		Assert.assertTrue(
+			url.contains("fileEntryId=" + _fileEntry.getFileEntryId()));
+		Assert.assertTrue(
+			url.contains(
+				"mvcRenderCommandName=" +
+					HtmlUtil.escapeURL("/document_library/view_file_entry")));
+		Assert.assertTrue(url.contains("version=" + _fileEntry.getVersion()));
+	}
+
+	@Test
+	public void testIsShow() throws Exception {
```

Comment:
What do you think about adding a method to test when this action is not show?

Reply:
@cgoncas, it's always shown, that's why I only created one test

Reply:
awesoime